src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -2306,6 +2308,85 @@ namespace Idint {
+    }
+
+    static ASR::expr_t *eval_Ishftc(Allocator &al, const Location &loc,
+            ASR::ttype_t* t1, Vec<ASR::expr_t*> &args) {
+        int64_t val = ASR::down_cast<ASR::IntegerConstant_t>(args[0])->m_n;
+        int64_t shift = ASR::down_cast<ASR::IntegerConstant_t>(args[1])->m_n;
+        int kind = 8*ASRUtils::extract_kind_from_ttype_t(ASR::down_cast<ASR::IntegerConstant_t>(args[0])->m_type);
+        int result;
+        if (shift < 0) {
+            shift = -shift;  
+            result = (val >> shift) | (val << (kind - shift));
+        } else {
+            result = (val << shift) | (val >> (kind - shift)); 
+        }
+        return make_ConstantWithType(make_IntegerConstant_t, result, t1, loc);
+    }
+
+    static inline ASR::asr_t* create_Ishftc(Allocator& al, const Location& loc,
+            Vec<ASR::expr_t*>& args,
+            const std::function<void (const std::string &, const Location &)> err) {
+        if (args.size() != 2) {
+            err("Intrinsic `Ishftc` function accepts exactly 2 arguments", loc);
+        }
+        ASR::ttype_t *type1 = ASRUtils::expr_type(args[0]);
+        ASR::ttype_t *type2 = ASRUtils::expr_type(args[1]);
+        if (!ASRUtils::is_integer(*type1) || !ASRUtils::is_integer(*type2)) {
+            err("Arguments of the `Ishftc` function must be Integer",
+                args[0]->base.loc);
+        }
+        ASR::expr_t *m_value = nullptr;
+        if (all_args_evaluated(args)) {
+            Vec<ASR::expr_t*> arg_values; arg_values.reserve(al, 2);
+            arg_values.push_back(al, expr_value(args[0]));
+            arg_values.push_back(al, expr_value(args[1]));
+            int64_t shift = ASR::down_cast<ASR::IntegerConstant_t>(arg_values[1])->m_n;
+            int kind = ASRUtils::extract_kind_from_ttype_t(ASR::down_cast<ASR::IntegerConstant_t>(arg_values[0])->m_type);
+            int bits = 8*kind;
+            if (bits < shift) {
+                err("The absolute value of SHIFT argument must be less than or equal to BIT_SIZE('I')",
+                args[0]->base.loc);
+            }
+            m_value = eval_Ishftc(al, loc, expr_type(args[0]), arg_values);
+        }
+        return ASR::make_IntrinsicScalarFunction_t(al, loc,
+            static_cast<int64_t>(IntrinsicScalarFunctions::Ishftc),
+            args.p, args.n, 0, ASRUtils::expr_type(args[0]), m_value);
+    }
+
+    static inline ASR::expr_t* instantiate_Ishftc(Allocator &al, const Location &loc,
+            SymbolTable *scope, Vec<ASR::ttype_t*>& arg_types, ASR::ttype_t *return_type,
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        declare_basic_variables("_lcompilers_Ishftc_" + type_to_str_python(arg_types[0]));
+        fill_func_arg("x", arg_types[0]);
+        fill_func_arg("y", arg_types[1]);
+        auto result = declare(fn_name, return_type, ReturnVar);
+        int kind = 8*ASRUtils::extract_kind_from_ttype_t(arg_types[0]);
```

Comment:
```suggestion
        int kind = ASRUtils::extract_kind_from_ttype_t(arg_types[0]);
        int no_of_bits = 8 * kind;
```

and then reuse `no_of_bits` (or `bitsize` or whichever name is more appropriate).